### PR TITLE
Fixed DoctrineResource::findEntity() criteria #171

### PR DESCRIPTION
**Pull request by thomasvargiu**
Related to issue #171

If the route parameter was stripped off of ```stripRouteParameterSuffix```, it will not be added in the criteria if match in ```$fieldNames```.

### CODE REVIEW

**Comment by TomHAnderson** [comment]
I'm fine with this patch but it needs a unit test to show it in action.

**Comment by thomasvargiu** [comment]
Of course. I'll do it asap. 
I was thinking that what I've done it's not enough, it doesn't cover another possible problem, so I need to do much work.

**Comment by thomasvargiu** [comment]
Added a "parent" relation to the test ```Entity\Album``` with ```album``` name. Without my patch, tests fails.

My patch does:
- Remove the ```route_identifier_name``` from criteria, because it's already added, and it's not possible to add another criteria with the same ```route_identifier_name```. This resolve issue #171.
- If a route param contains something like ```:message_id```, the criteria ```['message' => ':message_id'``` will not be added if key ```message``` is in field names and not in association names.

Tests updated with a self-reference association in the ```Album``` entity. Without my patch, tests fails.
I've done some little refactoring and fixed some annotations.

File: src/Server/Resource/DoctrineResourceFactory.php [hunk 1]
```diff
@@ -109,6 +109,7 @@ public function createServiceWithName(ServiceLocatorInterface $serviceLocator, $
         $config = $serviceLocator->get('Config');
         $doctrineConnectedConfig = $config['zf-apigility']['doctrine-connected'][$requestedName];
 
+        $restConfig = null;
         foreach ($config['zf-rest'] as $restControllerConfig) {
             if ($restControllerConfig['listener'] == $requestedName) {
                 $restConfig = $restControllerConfig;
```
Comment (TomHAnderson) [inline src/Server/Resource/DoctrineResourceFactory.php]: Once found do a `break;`

Comment (thomasvargiu) [inline src/Server/Resource/DoctrineResourceFactory.php]: @TomHAnderson  There is already a ```break;```. View the whole file.

**Comment by thomasvargiu** [comment]
@TomHAnderson  Are you fine with this patch?

File: src/Server/Resource/DoctrineResource.php [hunk 2]
```diff
@@ -660,21 +686,24 @@ protected function findEntity($id, $method)
...
                 $routeMatchParam,
-                (-1 * abs(strlen($this->getStripRouteParameterSuffix())) == $this->getStripRouteParameterSuffix())
+                -1 * strlen($this->getStripRouteParameterSuffix())
             )) {
-                $routeMatchParam = substr(
-                    $routeMatchParam,
-                    0,
-                    strlen($routeMatchParam) - strlen($this->getStripRouteParameterSuffix())
-                );
+                $routeMatchParam = substr($routeMatchParam, 0, -1 * strlen($this->getStripRouteParameterSuffix()));
```
Comment (TomHAnderson) [inline src/Server/Resource/DoctrineResource.php]: With a unit test for lines 701-703 & 708 I'll merge this.

**Comment by thomasvargiu** [comment]
I edited the patch. I remove just the route identifier name from the criteria, it's enought to fix issue #171 without creating BC breaks. I also made a change on tests, to update them. I created a route like ```/test/rest[/artist/:artist_id]/album[/:album_id]```. This is what should be an rpc service, but an user could be that.

**Comment by TomHAnderson** [comment]
merged to develop

**Pull request closed without merging**